src/Crawler.php

```diff
@@ -561,37 +443,24 @@ protected function maximumCrawlCountReached(): bool
         return $this->crawledUrlCount >= $this->maximumCrawlCount;
     }
 
-    protected function mayIndex(RobotsHeaders  $robotsHeaders, RobotsMeta $robotsMeta): bool
+    protected function getFulfilledHandler(): CrawlRequestFulfilled
     {
-        if (! $this->respectRobots) {
-            return true;
-        }
-
-        if (! $robotsHeaders->mayIndex()) {
-            return false;
-        }
-
-        if (! $robotsMeta->mayIndex()) {
-            return false;
-        }
-
-        return true;
+        return new CrawlRequestFulfilled(
```

Comment:
I'd inject only the crawler itself here instead of all these. That way we don't have to touch this function when extra stuff needs to be passed to `CrawlRequestFulfilled` in the future.